### ISSUE TITLE
[Feature Request] In place batch normalisation +1% AP Top1

### ISSUE THREAD

**Issue by LukeAI** [AlexeyAB/darknet#3755]
In-Place Activated BatchNorm (InPlace-ABN) is a novel approach to reduce the memory required for training deep networks. It allows for up to 50% memory savings in modern architectures such as ResNet, ResNeXt and Wider ResNet by redefining BN + non linear activation as a single in-place operation, while smartly dropping or recomputing intermediate buffers as needed.
https://arxiv.org/abs/1712.02616
https://github.com/mapillary/inplace_abn

**Comment by AlexeyAB** [comment]
But we should implement Reverse-function for every Activation.
Perhaps there are still some pitfalls.
